src/soundLoop.js

```diff
@@ -0,0 +1,228 @@
+'use strict';
+
+define(function (require) {
+  var p5sound = require('master');
+  var Clock = require('Tone/core/Clock');
+
+  /**
+   * SoundLoop
+   *
+   * @class p5.SoundLoop
+   * @constructor
+   */
+  p5.SoundLoop = function(callback, interval, BPM) {
+
+    this.callback = callback;
+
+    this.musicalTimeMode = typeof this._interval === 'number' ? false : true;
+    this._interval = interval;
+
+    //These variables should only be modified if using musicalTimeMode
+    //If these variables are modified when interval is measured in seconds,
+    //the interval will become inaccurate. 
+    //ex. 8 second interval at 60BPM in 4/4 time will be 8 seconds long
+    //8 second interval at 120BPM in 5/4 time will be 3.2 seconds long
+    this._timeSignature = 4;
+    this._bpm = BPM || 60;
+
+    this.isPlaying = false;
+    var self = this;
+    this.clock = new Clock({
+      'callback' : function(time) {
+        var timeFromNow = time - p5sound.audiocontext.currentTime;
+        if (timeFromNow > 0) {self.callback(timeFromNow);}
```

Comment:
why would this be less than zero anyway? ah, @jvntf and I are testing together and found that this is the case when we load the page. Maybe worth adding a comment?